src/script/descriptor.cpp

```diff
@@ -19,6 +19,117 @@
 
 namespace {
 
+////////////////////////////////////////////////////////////////////////////
+// Checksum                                                               //
+////////////////////////////////////////////////////////////////////////////
+
+// This section implements a checksum algorithm for descriptors with the following properties:
+// * Every 1 character substitution error counts as 1 or 2 symbol errors, but:
+//   * An error substituting a character from 0123456789()[],'/*abcdefgh@:$%{} for another in
```

Comment:
Maybe flip this around, and explain a bit more how symbol error count works:

```c++
// * Mistakes in a descriptor string are measured in symbol errors. A higher symbol
//   error count is more difficult to detect, because it becomes indistinguishable
//   from a different descriptor.
// * An error substituting a character from 0123456789()[],'/*abcdefgh@:$%{} for
//   another in that set always counts as 1 symbol error.
// * A case error always counts as 1 symbol error.
// * Any other 1 character substitution error counts as 1 or 2 symbol errors.
// * Note that hex encoded keys are covered by these special characters, whereas xprivs
//   and xpubs use different characters, but already have their own checksum mechanism.
//   Function names like "multi()" use different characters, but mistakes would generally
//   result in an unparseable descriptor. 
```

Some of this is also explained in `DescriptorChecksum` in different wording.

Reply:
Cool, that's more clear. I've included it with some copy-editing.